flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveMetastoreClientWrapper.java

```diff
@@ -210,6 +212,11 @@ public boolean updatePartitionColumnStatistics(ColumnStatistics columnStatistics
 
 	//-------- Start of shimmed methods ----------
 
+	public Set<String> getNotNullColumns(Configuration conf, String dbName, String tableName) {
+		HiveShim hiveShim = HiveShimLoader.loadHiveShim(hiveVersion);
```

Comment:
no need to create a HiveShim instance every time. HiveMetastoreClientWrapper can just keep a HiveShim instance built in constructor?